model/buildlogger_storage.go

```diff
@@ -0,0 +1,41 @@
+package model
+
+import (
+	"github.com/evergreen-ci/pail"
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/pkg/errors"
+)
+
+type LogFormat string
+
+const (
+	LogFormatUnknown LogFormat = "unknown"
+	LogFormatText    LogFormat = "text"
+	LogFormatJSON    LogFormat = "json"
+	LogFormatBSON    LogFormat = "bson"
+)
+
+func (lf LogFormat) Validate() error {
+	switch lf {
+	case LogFormatUnknown, LogFormatText, LogFormatJSON, LogFormatBSON:
```

Comment:
what does it mean to have an unknown log format? 

are there schemas associated with json/bson that we'd want to track which would mean that we'd need more information?

Reply:
I guess unknown would be the default if a log format is not specified, similar to how we have an unknown format for perf artifacts. Not sure what you mean about the schemas associated with json/bson, could you give an example?